docs/protocol/network/data-structures/transaction.md

```diff
@@ -0,0 +1,136 @@
+| `commitment` | `[u8]` | Signed hash of the data request result by an individual node                                               |
+| `value`      | `u64`  | Remaining transaction value that will be used as reward to be distributed after consensus has been reached |
+
+The `value` of the commit output depends on the number of committers:
+
+```math
+commit_value = (data_request_value / witnesses) - commit_fee
+```
+
+### Reveal outputs
+
+| Field   | Type  | Description                                                                                                |
+| ------- | :---: | ---------------------------------------------------------------------------------------------------------- |
+| `value` | `u64` | Remaining transaction value that will be used as reward to be distributed after consensus has been reached |
+
+The `value` of the reveal output depends on the number of committers that revealed their data request results:
+
+```math
+reveal_value = commit_value - reveal_fee
+```
+
+## Claims
+
+As aforementioned, there should be a claim for each input, which provides an spending condition in order to consume an unspent output (UTXO).
+
+In Witnet, different output types have specific claims:
+
+- P2PKH claim: demostrate ownership of private key that signs the hash of the transaction.
+- Data request claim: provide Proof of Eligibility (PoE) and include Public Key as sign of commitment.
+- Commit claim: provide a signed reveal message matching a previous `commitment` (hash) and that matches with the previous advertised public key.
+- Reveal claim: provide a data request consensus result which is valid according to the consensus script and previous reveals outputs.
+
+### P2PKH claim
+
+| Field        |  Type  | Description                                     |
+| ------------ | :----: | ----------------------------------------------- |
+| `signature`  | `[u8]` | Signed hash of the transaction                  |
+| `public_key` | `[u8]` | Public Key of the P2PKH outpoint to be consumed |
+
+### Data request claim (**commit phase**)
+
+A **data request** output has to be consumed/used by a number of **witnesses**, i.e. there will be as many claims as **witnesses** have been defined.
+
+| Field        |  Type  | Description                           |
+| ------------ | :----: | ------------------------------------- |
+| `poe`        | `[u8]` | Proof of eligibility                  |
+| `public_key` | `[u8]` | Public Key used for computing the PoE |
+
+### Commit claim (**reveal phase**)
+
+| Field    |  Type  | Description                                                                                                |
+| -------- | :----: | ---------------------------------------------------------------------------------------------------------- |
+| `reveal` | `[u8]` | Signed data request result using the previously advertised public key during the previous **commit phase** |
```

Comment:
This is where the nonce used in the commitment should be published.

Reply:
I think so... i propose to add a `nonce` field of type `u32`. WDYT?